octoprint_mrbeam/analytics/value_collector.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from octoprint_mrbeam.mrb_logger import mrb_logger
+
+
+class ValueCollector(object):
+	def __init__(self, name):
+		self.name = name
+		self.valueList = list()
+		# self._logger = mrb_logger("octoprint.plugins.mrbeam.analyticshandler")
+
+	def addValue(self, value):
+		self.valueList.append(value)
+
+	def getSummary(self):
+		"""
+		Returns a dict with all the dust statistics
+		(mean,median,count,std,)
+		:param valueList:
+		:return:
+		"""
+		count = len(self.valueList)
+		# self._logger.debug('Collector <{}> has values: {}'.format(self.name,self.valueList))
+
+		if count > 0:
+			arr = np.asarray(self.valueList)
+			descDict = {
+				'median': np.median(arr),
+				'mean': np.mean(arr),
+				'min': min(self.valueList),
+				'max': max(self.valueList),
+				'25p': np.percentile(arr,25),
+				'75p': np.percentile(arr,75),
+				'std': np.std(arr),
+				'count': len(self.valueList)
+			}
+		else:
+			descDict = {'count':count}
+
+		# make all values float for json.dump()-compability
```

Comment:
Would it make a difference, if we'd used something like JSON.stringify() ? Probably not...

Reply:
hm, but then we still need to round it, because otherwise we would have around 10+ digits after the comma.